### ISSUE TITLE
Add support for Arcadyan KVD21 Gateway

### ISSUE THREAD

**Issue by highvolt-dev** [kclejeune/TMobile-ISP-Client#12]
Now that there is more than just the Nokia Gateway, it'd be good to support T-Mobile's other hardware models.

Here's some information I've gathered: https://github.com/highvolt-dev/tmo-monitor/issues/22

I have additional info beyond what is documented there as well if you need anything.

**Comment by kclejeune** [comment]
Yes, I was wondering about this - do you have one of these? I'm thinking it's probably best if we pull out the api abstraction into a separate dependency so that we can write type definitions that can be compatible with both.

**Comment by highvolt-dev** [comment]
Yes, it would be ideal to abstract it so the data structure can be normalized.

I don't have an Arcadyan gateway, just the Nokia. I had to guide someone to get the info I have, but I have sample requests and responses for various endpoints.

**Comment by kclejeune** [comment]
> Yes, it would be ideal to abstract it so the data structure can be normalized.
> 
> 
> 
> I don't have an Arcadyan gateway, just the Nokia. I had to guide someone to get the info I have, but I have sample requests and responses for various endpoints.

Got it. I'd be happy to pull this into a typescript library that exposes types for responses on different devices, and some utilities to convert the direct responses into a new common format. 

If you can get those sample responses to me (kennan@case.edu, kclejeune#3453 on discord, or something else) I can work on incorporating the new device into this too.
